<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1073</width>
    <height>642</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Melody Query Tool</string>
  </property>
  <property name="windowIcon">
   <iconset>
    <normaloff>icons/music.png</normaloff>icons/music.png</iconset>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QHBoxLayout" name="horizontalLayout_4">
    <item>
     <layout class="QVBoxLayout" name="verticalLayout_6">
      <item>
       <widget class="QLabel" name="label_2">
        <property name="font">
         <font>
          <pointsize>10</pointsize>
         </font>
        </property>
        <property name="text">
         <string>Database items</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
       </widget>
      </item>
      <item alignment="Qt::AlignLeft">
       <widget class="QListWidget" name="database_item_list_widget">
        <property name="minimumSize">
         <size>
          <width>200</width>
          <height>0</height>
         </size>
        </property>
        <property name="editTriggers">
         <set>QAbstractItemView::NoEditTriggers</set>
        </property>
        <property name="selectionBehavior">
         <enum>QAbstractItemView::SelectRows</enum>
        </property>
        <property name="modelColumn">
         <number>0</number>
        </property>
        <property name="selectionRectVisible">
         <bool>true</bool>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <layout class="QVBoxLayout" name="verticalLayout_3">
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout">
        <item>
         <layout class="QVBoxLayout" name="verticalLayout">
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_6">
            <item>
             <layout class="QVBoxLayout" name="verticalLayout_10">
              <item>
               <layout class="QHBoxLayout" name="horizontalLayout_3">
                <item>
                 <spacer name="horizontalSpacer">
                  <property name="orientation">
                   <enum>Qt::Horizontal</enum>
                  </property>
                  <property name="sizeHint" stdset="0">
                   <size>
                    <width>40</width>
                    <height>20</height>
                   </size>
                  </property>
                 </spacer>
                </item>
                <item>
                 <widget class="QLabel" name="query_name_label">
                  <property name="font">
                   <font>
                    <pointsize>10</pointsize>
                   </font>
                  </property>
                  <property name="text">
                   <string>Please select a query</string>
                  </property>
                  <property name="alignment">
                   <set>Qt::AlignCenter</set>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
              <item>
               <layout class="QHBoxLayout" name="horizontalLayout_2">
                <item>
                 <spacer name="horizontalSpacer_2">
                  <property name="orientation">
                   <enum>Qt::Horizontal</enum>
                  </property>
                  <property name="sizeHint" stdset="0">
                   <size>
                    <width>40</width>
                    <height>20</height>
                   </size>
                  </property>
                 </spacer>
                </item>
                <item>
                 <widget class="QToolButton" name="open_query_button">
                  <property name="toolTip">
                   <string>Open a query</string>
                  </property>
                  <property name="text">
                   <string>...</string>
                  </property>
                  <property name="icon">
                   <iconset>
                    <normaloff>icons/open.png</normaloff>icons/open.png</iconset>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QToolButton" name="save_query_button">
                  <property name="toolTip">
                   <string>Save a recorded query</string>
                  </property>
                  <property name="text">
                   <string>...</string>
                  </property>
                  <property name="icon">
                   <iconset>
                    <normaloff>icons/music.png</normaloff>icons/music.png</iconset>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QToolButton" name="play_query_button">
                  <property name="text">
                   <string>...</string>
                  </property>
                  <property name="icon">
                   <iconset>
                    <normaloff>icons/play-record.png</normaloff>icons/play-record.png</iconset>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QToolButton" name="record_query_button">
                  <property name="toolTip">
                   <string>record a new query</string>
                  </property>
                  <property name="text">
                   <string>...</string>
                  </property>
                  <property name="icon">
                   <iconset>
                    <normaloff>icons/record.png</normaloff>icons/record.png</iconset>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QToolButton" name="search_query_button">
                  <property name="toolTip">
                   <string>Start a search for the selected query</string>
                  </property>
                  <property name="text">
                   <string>...</string>
                  </property>
                  <property name="icon">
                   <iconset>
                    <normaloff>icons/search.png</normaloff>icons/search.png</iconset>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
             </layout>
            </item>
           </layout>
          </item>
          <item>
           <spacer name="verticalSpacer">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>20</width>
              <height>40</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <layout class="QVBoxLayout" name="verticalLayout_11">
            <item>
             <layout class="QVBoxLayout" name="verticalLayout_12">
              <item>
               <layout class="QHBoxLayout" name="horizontalLayout_7">
                <item alignment="Qt::AlignLeft">
                 <widget class="QLabel" name="label_3">
                  <property name="font">
                   <font>
                    <pointsize>10</pointsize>
                   </font>
                  </property>
                  <property name="text">
                   <string>Currently Playing:</string>
                  </property>
                  <property name="textFormat">
                   <enum>Qt::AutoText</enum>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QLabel" name="currently_playing_label">
                  <property name="font">
                   <font>
                    <pointsize>10</pointsize>
                   </font>
                  </property>
                  <property name="text">
                   <string/>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
              <item>
               <widget class="QSlider" name="music_position_slider">
                <property name="maximumSize">
                 <size>
                  <width>16777215</width>
                  <height>10</height>
                 </size>
                </property>
                <property name="styleSheet">
                 <string notr="true">QSlider::groove:horizontal {
border: 1px solid #bbb;
background: white;
height: 4px;
border-radius: 2px;
}

QSlider::sub-page:horizontal {
background: qlineargradient(x1: 0, y1: 0,    x2: 0, y2: 1,
    stop: 0 #55f, stop: 1 #55f);
background: qlineargradient(x1: 0, y1: 0.2, x2: 1, y2: 1,
    stop: 0 #55f, stop: 1 #55f);
border: 1px solid #777;
height: 10px;
border-radius: 2px;
}

QSlider::add-page:horizontal {
background: #fff;
border: 1px solid #777;
height: 10px;
border-radius: 2px;
}

QSlider::handle:horizontal {
background: qlineargradient(x1:0, y1:0, x2:1, y2:1,
    stop:0 #eee, stop:1 #ccc);
border: 1px solid #777;
width: 10px;
margin-top: -2px;
margin-bottom: -2px;
border-radius: 2px;
}

QSlider::handle:horizontal:hover {
background: qlineargradient(x1:0, y1:0, x2:1, y2:1,
    stop:0 #fff, stop:1 #ddd);
border: 1px solid #444;
border-radius: 2px;
}

QSlider::sub-page:horizontal:disabled {
background: #bbb;
border-color: #999;
}

QSlider::add-page:horizontal:disabled {
background: #eee;
border-color: #999;
}

QSlider::handle:horizontal:disabled {
background: #eee;
border: 1px solid #aaa;
border-radius: 4px;
}</string>
                </property>
                <property name="orientation">
                 <enum>Qt::Horizontal</enum>
                </property>
               </widget>
              </item>
              <item>
               <layout class="QHBoxLayout" name="horizontalLayout_8">
                <item>
                 <spacer name="horizontalSpacer_3">
                  <property name="orientation">
                   <enum>Qt::Horizontal</enum>
                  </property>
                  <property name="sizeHint" stdset="0">
                   <size>
                    <width>40</width>
                    <height>20</height>
                   </size>
                  </property>
                 </spacer>
                </item>
                <item>
                 <widget class="QToolButton" name="stop_button">
                  <property name="text">
                   <string>...</string>
                  </property>
                  <property name="icon">
                   <iconset>
                    <normaloff>icons/stop.png</normaloff>icons/stop.png</iconset>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QToolButton" name="play_pause_button">
                  <property name="text">
                   <string>...</string>
                  </property>
                  <property name="icon">
                   <iconset>
                    <normaloff>icons/play.png</normaloff>icons/play.png</iconset>
                  </property>
                 </widget>
                </item>
                <item>
                 <spacer name="horizontalSpacer_4">
                  <property name="orientation">
                   <enum>Qt::Horizontal</enum>
                  </property>
                  <property name="sizeHint" stdset="0">
                   <size>
                    <width>40</width>
                    <height>20</height>
                   </size>
                  </property>
                 </spacer>
                </item>
               </layout>
              </item>
             </layout>
            </item>
           </layout>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QVBoxLayout" name="verticalLayout_5">
          <item alignment="Qt::AlignTop">
           <widget class="QLabel" name="label">
            <property name="font">
             <font>
              <pointsize>10</pointsize>
             </font>
            </property>
            <property name="text">
             <string>Query results</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
           </widget>
          </item>
          <item alignment="Qt::AlignRight">
           <widget class="QTableWidget" name="result_table">
            <property name="enabled">
             <bool>true</bool>
            </property>
            <property name="editTriggers">
             <set>QAbstractItemView::NoEditTriggers</set>
            </property>
            <property name="dragDropOverwriteMode">
             <bool>false</bool>
            </property>
            <property name="selectionMode">
             <enum>QAbstractItemView::SingleSelection</enum>
            </property>
            <property name="selectionBehavior">
             <enum>QAbstractItemView::SelectRows</enum>
            </property>
            <property name="showGrid">
             <bool>false</bool>
            </property>
            <property name="wordWrap">
             <bool>false</bool>
            </property>
            <property name="cornerButtonEnabled">
             <bool>false</bool>
            </property>
            <attribute name="horizontalHeaderCascadingSectionResizes">
             <bool>false</bool>
            </attribute>
            <attribute name="horizontalHeaderDefaultSectionSize">
             <number>54</number>
            </attribute>
            <attribute name="horizontalHeaderHighlightSections">
             <bool>false</bool>
            </attribute>
            <attribute name="horizontalHeaderMinimumSectionSize">
             <number>20</number>
            </attribute>
            <attribute name="horizontalHeaderStretchLastSection">
             <bool>true</bool>
            </attribute>
            <attribute name="verticalHeaderVisible">
             <bool>false</bool>
            </attribute>
            <row>
             <property name="text">
              <string>Neue Zeile</string>
             </property>
            </row>
            <row>
             <property name="text">
              <string>Neue Zeile</string>
             </property>
            </row>
            <row>
             <property name="text">
              <string>Neue Zeile</string>
             </property>
            </row>
            <column>
             <property name="text">
              <string>Rank</string>
             </property>
            </column>
            <column>
             <property name="text">
              <string>Match at</string>
             </property>
            </column>
            <column>
             <property name="text">
              <string>Music Piece</string>
             </property>
            </column>
            <item row="0" column="0">
             <property name="text">
              <string>sdfa</string>
             </property>
            </item>
            <item row="1" column="0">
             <property name="text">
              <string>af</string>
             </property>
            </item>
            <item row="1" column="2">
             <property name="text">
              <string>sdf</string>
             </property>
            </item>
            <item row="2" column="1">
             <property name="text">
              <string>asdf</string>
             </property>
            </item>
            <item row="2" column="2">
             <property name="text">
              <string>sdfa</string>
             </property>
            </item>
           </widget>
          </item>
          <item alignment="Qt::AlignRight|Qt::AlignTop">
           <widget class="QListWidget" name="query_result_list_widget">
            <property name="minimumSize">
             <size>
              <width>200</width>
              <height>0</height>
             </size>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </item>
      <item alignment="Qt::AlignVCenter">
       <widget class="PianoWidget" name="midiViewer" native="true"/>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_5">
        <item>
         <spacer name="horizontalSpacer_6">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <widget class="QSlider" name="zoom_slider">
          <property name="minimum">
           <number>10</number>
          </property>
          <property name="maximum">
           <number>200</number>
          </property>
          <property name="singleStep">
           <number>1</number>
          </property>
          <property name="value">
           <number>50</number>
          </property>
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="tickInterval">
           <number>1</number>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QToolButton" name="zoom_out_button">
          <property name="text">
           <string>...</string>
          </property>
          <property name="icon">
           <iconset>
            <normaloff>icons/zoom-out.png</normaloff>icons/zoom-out.png</iconset>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QToolButton" name="zoom_in_button">
          <property name="text">
           <string>...</string>
          </property>
          <property name="icon">
           <iconset>
            <normaloff>icons/zoom-in.png</normaloff>icons/zoom-in.png</iconset>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="horizontalSpacer_5">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1073</width>
     <height>21</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuData">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="action_open_database"/>
    <addaction name="separator"/>
    <addaction name="action_open_query"/>
    <addaction name="action_save_query"/>
    <addaction name="separator"/>
    <addaction name="actionExit"/>
   </widget>
   <addaction name="menuData"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="actionLoad_Database">
   <property name="icon">
    <iconset>
     <normaloff>icons/open.png</normaloff>icons/open.png</iconset>
   </property>
   <property name="text">
    <string>Load Database</string>
   </property>
  </action>
  <action name="actionExit">
   <property name="text">
    <string>Exit</string>
   </property>
  </action>
  <action name="action_save_query">
   <property name="icon">
    <iconset>
     <normaloff>ui/save-as.png</normaloff>ui/save-as.png</iconset>
   </property>
   <property name="text">
    <string>Save Query...</string>
   </property>
  </action>
  <action name="action_load_query">
   <property name="icon">
    <iconset>
     <normaloff>ui/music.png</normaloff>ui/music.png</iconset>
   </property>
   <property name="text">
    <string>Query</string>
   </property>
  </action>
  <action name="action_load_database">
   <property name="icon">
    <iconset>
     <normaloff>ui/database.png</normaloff>ui/database.png</iconset>
   </property>
   <property name="text">
    <string>Database</string>
   </property>
  </action>
  <action name="action_open_database">
   <property name="icon">
    <iconset>
     <normaloff>ui/database.png</normaloff>ui/database.png</iconset>
   </property>
   <property name="text">
    <string>Open Database...</string>
   </property>
  </action>
  <action name="action_open_query">
   <property name="icon">
    <iconset>
     <normaloff>ui/open.png</normaloff>ui/open.png</iconset>
   </property>
   <property name="text">
    <string>Open Query...</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>PianoWidget</class>
   <extends>QWidget</extends>
   <header>piano_roll_editor</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
